### PR TITLE
Fix of #493: No way to turn-off Hardware acceleration in AuthenticationActivity WebView

### PR DESCRIPTION
**Pull request by talk2arpit**
Issue: By default WebView is always hardwareAccelerated due to which on
some devices it crashes intermittently because of bad GPU drivers. There
is a way to turn-off hardware acceleration by setting webview layer type
to 'LAYER_TYPE_SOFTWARE' but as webView used in AuthenticationActivity
and AuthenticationDialog is not directly accessible to caller so there
is no way caller could disable it.

Changes: Added a method in AuthenticationSettings i.e.
setHardwareAcceleration, which caller can use to disable WebView
hardware acceleration.

### CODE REVIEW

File: src/src/com/microsoft/aad/adal/AuthenticationSettings.java [hunk 1]
```diff
@@ -253,4 +255,31 @@ public void setReadTimeOut(int timeOutMillis) {
...
+     * {@link AuthenticationActivity} and {@link AuthenticationDialog}. 
+     * By default hardware acceleration is enable in WebView.
+     *  
+     * @param enable
+     *             if true, WebView would be hardwareAccelerated else it 
+     *             would be disable.
+     *             
+     * @see #isWebViewHardwareAccelerated()
+     */
+    public void setWebViewHardwareAcceleration(boolean enable) {
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationSettings.java]: What you really want to achieve is to disable the webiew acceleration. I would like to call this disableWebviewHardwareAcceleration.

Comment (talk2arpit) [inline src/src/com/microsoft/aad/adal/AuthenticationSettings.java]: By default, hardware acceleration is always enable in View, which can be disabled. [ref: http://developer.android.com/guide/topics/graphics/hardware-accel.html ]. Due to hardware accelerated AuthenticationActivity WebView, we have seen couple of crashes on some devices so we want to disable it in our app.
Here, this new method can be used to disable webview hardware acceleration.  Although by renaming this method to disableWebViewHardwareAccelertaion, would also solve the purpose but for completeness purpose i.e. enable/disable and query hardware acceleration in a same session, I use this name, something similar to setSkipBorker.

Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationSettings.java]: Can we have the the name as setEnableHardwareAcceleration(boolean enabled)/getEnableHardwareAcceleration?

Comment (RandalliLama) [inline src/src/com/microsoft/aad/adal/AuthenticationSettings.java]: This sounds too much like it is changing the default.  The current default is to use hardware acceleration and that should remain the default.  The caller should have a flag they can set to true to disable hardware acceleration.  That would imply the name should be something like:

setDisableWebViewHardwareAcceleration
getDisableWebViewHardwareAcceleration

Comment (talk2arpit) [inline src/src/com/microsoft/aad/adal/AuthenticationSettings.java]: I have renamed the methods as suggested. Kindly review the latest changes.

**Comment by addev-ashish** [comment]
#493

**Pull request merged**